gnovm/pkg/gnolang/preprocess.go

```diff
@@ -3039,6 +3042,18 @@ func checkIntegerType(xt Type) {
 	}
 }
 
+func checkAssignmentMismatch(d Decl) {
+	if cd, ok := d.(*ValueDecl); ok {
+		numNames := len(cd.NameExprs)
+		numValues := len(cd.Values)
+		if numValues > 0 && numValues != numNames {
```

Comment:
can you also consider this case:
```go
package main

func foo() (int, bool) {
	return 1, true
}

func main() {

	var a, b = 2, foo()

	println(a, b)

}
```

Reply:
This case cannot be checked here because we need to evaluate `evalStaticTypeOfRaw(store, last, cx)` to determine the number of elements that the call returns. This function requires preprocessing the call before.

Reply:
can we do it in trans_leave? (I'm not sure).

Reply:
just leave it.  I guess it probably been fixed in another PR. thank u.

Reply:
Don't know if it is good to do it in this PR.
But done here: https://github.com/gnolang/gno/pull/2206/commits/564735c4855b43fc9f6d7634aa7cd47904d7e6c1

Reply:
Added here:https://github.com/gnolang/gno/commit/f9935a8b89c4bc5dee465cdc3affa801f68eb783

But don't know if we should do it in this PR